### ISSUE TITLE
[Bug]: same job running twice at the closely time

### ISSUE THREAD

**Issue by zshnb** [taskforcesh/bullmq#1900]
### Version

v3.10.1

### Platform

NodeJS

### What happened?

A bug happened!
the job with same job id running in the same worker twice at the closely time, i add job only once. is there some wrong in my code
```typescript
// add job
const jobId = `${payload.importTaskId}:split-audio:${genNanoId(2)}`
await sourceQueue.add('split-audio', payload, {
  ...jobOptions,
  jobId,
  attempts,
})
```
![image](https://github.com/taskforcesh/bullmq/assets/17894980/8c0b7459-488b-4918-8de2-b56c190ef4e2)
![image](https://github.com/taskforcesh/bullmq/assets/17894980/13801938-b9e5-4f52-8591-26f447e41309)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by roggervalf** [comment]
Are those registers active at the same time? it would be nice to have a way to reproduce

**Comment by zshnb** [comment]
> Are those registers active at the same time? it would be nice to have a way to reproduce

no i only have one worker, and isn't easy to reproduce

**Comment by manast** [comment]
I see you are using custom job ids, are you sure you are keeping enough completed jobs so that duplicate ids are ignored?

**Comment by zshnb** [comment]
> I see you are using custom job ids, are you sure you are keeping enough completed jobs so that duplicate ids are ignored?

yes, i'm sure these two job id in pic is same

**Comment by manast** [comment]
You did not answer to my question: "are you sure you are keeping enough completed jobs so that duplicate ids are ignored?"

**Comment by zshnb** [comment]
> You did not answer to my question: "are you sure you are keeping enough completed jobs so that duplicate ids are ignored?"

oh sorry, my config is remove completed job, but you can see the timestamp in pic, these two jobs were received by one worker and in running

**Comment by manast** [comment]
If you added the same job twice with the same ID and removeOnComplete is true, then this is the correct behavior.

**Comment by zshnb** [comment]
> If you added the same job twice with the same ID and removeOnComplete is true, then this is the correct behavior.

i only add once

**Comment by manast** [comment]
The thing is, it is not possible for the same job to exist 2 times as completed, it is just not possible. So I do not know what that screenshot is showing... If you can provide some code that reproduces the issue we can look deeper into it.

**Comment by zshnb** [comment]
> 

yes i think so, and i also can't write code to reproduce this case, in my application this is very occasionally, anyway thank you for answer my question
